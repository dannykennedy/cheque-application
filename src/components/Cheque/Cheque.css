.cheque {
    background: linear-gradient(rgba(68, 179, 230, 0.5), white);
    height: 300px;
    width: 90%;
    margin: auto;
    display: grid;
    grid-template-rows: 1fr 2fr 1fr;
    box-shadow: 0 4px 5px 0 rgba(0, 0, 0, 0.2);
    transition: 0.2s;
    border-radius: 2px;
    padding: 15px;
    margin-top: 50px;
}

.cheque:hover {
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.cheque-header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.cheque-logo-area {
    width: 250px;
    height: 100px;
}

.cheque-body {
    display: grid;
    grid-template-columns: 65% 35%;
}

.cheque-date-area {
    margin-top: 10px;
    display: flex;
    align-items: flex-start;
    flex-direction: column;
}

.cheque-date-row {
    display: grid;
    align-items: center;
    grid-template-columns: auto 1fr;
    column-gap: 10px;
}

.cheque-numeric-amount-area {
    display: grid;
    justify-items: center;
    grid-template-rows: 40px 1fr;
    box-sizing: border-box;
}

.cheque-numeric-amount {
    background-color: white;
    border: 2px solid #ccc;
    width: 80%;
    height: 100%;
    display: grid;
    align-items: center;
    padding-left: 10px;
    grid-template-columns: 1em 1fr;
}

.cheque-numeric-amount-text {
    height: fit-content;
}

.cheque-payee-name,
.cheque-date,
.cheque-text-amount,
.cheque-numeric-amount-text,
.cheque-signature-area {
    font-family: "adobe-handwriting-tiffany", cursive;
    color: black;
}

.cheque-payee-name,
.cheque-date,
.cheque-numeric-amount,
.cheque-signature-area {
    font-size: 1.2em;
}

.cheque-numeric-amount-text {
    font-size: 1em;
}

.cheque-signature-area {
    height: 1.5em;
}

.cheque-payee-name,
.cheque-date,
.cheque-text-amount,
.cheque-signature-area {
    border-bottom: 2px solid #000;
}

.cheque-payee-name,
.cheque-text-amount {
    border-right: 2px solid #000;
}

.cheque-payee-name {
    display: inline-block;
    width: 100%;
    margin-bottom: 10px;
}

.cheque-text {
    text-transform: uppercase;
}

.cheque-footer {
    display: grid;
}

.cheque-signature-area {
    justify-self: right;
}

.new-cheque-button-wrapper {
    width: 100%;
    display: grid;
    justify-content: center;
    margin-top: 50px;
}

@media (min-width: 800px) {
    .cheque {
        width: 80%;
        max-width: 700px;
    }
}
