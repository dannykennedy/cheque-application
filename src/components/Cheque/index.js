export { default } from "./Cheque.js";
