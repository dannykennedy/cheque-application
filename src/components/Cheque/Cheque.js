import React from "react";
import FontAwesome from "react-fontawesome";
import "./Cheque.css";
import ChequePic from "./BritishChequeAnnotated.png";
import { formatDate } from "../../modules/formatDate";
import SplitText from "react-pose-text";

// Full React Pose Text documentation can be found at
// https://popmotion.io/pose/api/react-pose-text
const charPoses = {
    exit: { opacity: 0, y: -2 },
    enter: {
        opacity: 1,
        y: 0,
        delay: ({ charIndex }) => charIndex * 50,
    },
};

function Cheque({ amount, amountInWords, payeeName, date, drawer }) {
    return (
        <div>
            <div className="cheque">
                <div className="cheque-header">
                    <div className="cheque-logo-area">
                        <div className="cheque-logo">
                            <div className="cheque-logo-icon">
                                <FontAwesome name="hand-holding-usd" />
                            </div>
                            <div className="cheque-bank-name">
                                First Bank of SmartCorp
                            </div>
                        </div>
                        <div className="cheque-branch-name cheque-text">
                            Castle Hill Branch
                        </div>
                    </div>
                    <div className="cheque-date-area">
                        <span className="cheque-text">Date: </span>
                        <span className="cheque-date">
                            <SplitText
                                initialPose="exit"
                                pose="enter"
                                charPoses={charPoses}
                            >
                                {formatDate(date)}
                            </SplitText>
                        </span>
                    </div>
                </div>
                <div className="cheque-body">
                    <div className="cheque-text-area">
                        <div className="cheque-text-payee-area">
                            <span className="cheque-text">Pay:</span>
                            <div className="cheque-payee-name">
                                <SplitText
                                    initialPose="exit"
                                    pose="enter"
                                    charPoses={charPoses}
                                >
                                    {payeeName}
                                </SplitText>
                            </div>
                        </div>
                        <div className="cheque-text-amount-area">
                            <div className="cheque-text-amount">
                                <span>
                                    <SplitText
                                        initialPose="exit"
                                        pose="enter"
                                        charPoses={charPoses}
                                    >
                                        {amountInWords + " only"}
                                    </SplitText>
                                </span>
                            </div>
                        </div>
                    </div>
                    <div className="cheque-numeric-amount-area">
                        <div className="cheque-numeric-amount">
                            <div>
                                <span>$</span>
                            </div>
                            <div className="cheque-numeric-amount-text">
                                <SplitText
                                    initialPose="exit"
                                    pose="enter"
                                    charPoses={charPoses}
                                >
                                    {amount.toString()}
                                </SplitText>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="cheque-footer">
                    <div className="cheque-signature-area">
                        <SplitText
                            initialPose="exit"
                            pose="enter"
                            charPoses={charPoses}
                        >
                            {drawer}
                        </SplitText>
                    </div>
                </div>
            </div>
            {/* <img src={ChequePic} alt="cheque" /> */}
        </div>
    );
}

export default Cheque;
