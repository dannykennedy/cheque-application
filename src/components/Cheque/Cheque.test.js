import renderer from "react-test-renderer";
import React from "react";
import { Cheque } from "./Cheque";

test("Tag snapshot test", () => {
    const component = renderer.create(
        <Cheque
            drawer={"John Jones"}
            amount={45}
            payee={"Dan Kennedy"}
            date={new Date()}
            testing={true}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
