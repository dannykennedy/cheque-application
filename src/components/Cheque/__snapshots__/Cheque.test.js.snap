// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheque snapshot test 1`] = `
<div>
  <div
    className="cheque"
  >
    <div
      className="cheque-header"
    >
      <div
        className="cheque-logo-area"
      >
        <div
          className="logo-wrapper"
        >
          <div
            className="logo-icon"
          >
            <span
              aria-hidden={true}
              className="fa fa-hand-holding-usd"
            />
          </div>
          <div
            className="logo-bank-name"
          >
            First Bank of SmartCorp
          </div>
        </div>
        <div
          className="cheque-branch-name cheque-text"
        >
          Castle Hill Branch
        </div>
      </div>
      <div
        className="cheque-date-area"
      >
        <div
          className="cheque-date-row"
        >
          <span
            className="cheque-text"
          >
            Date: 
          </span>
          <div
            className="cheque-date"
          >
            <div>
              <div>
                25/11/2019
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cheque-body"
    >
      <div
        className="cheque-text-area"
      >
        <div
          className="cheque-text-payee-area"
        >
          <span
            className="cheque-text"
          >
            Pay:
          </span>
          <div
            className="cheque-payee-name"
          >
            <div>
              <div>
                Dan Kennedy
              </div>
            </div>
          </div>
        </div>
        <div
          className="cheque-text-amount-area"
        >
          <div
            className="cheque-text-amount"
          >
            <span>
              <div>
                <img
                  alt="loading"
                  className="loading-notice-spinner"
                  height="20px"
                  src="./images/loading.gif"
                  style={
                    Object {
                      "alignSelf": "center",
                      "display": "inline-block",
                    }
                  }
                  width="auto"
                />
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        className="cheque-numeric-amount-area"
      >
        <div
          className="cheque-numeric-amount"
        >
          <div>
            <span>
              $
            </span>
          </div>
          <div
            className="cheque-numeric-amount-text"
          >
            <div>
              <div>
                45.00
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cheque-footer"
    >
      <div
        className="cheque-signature-area"
      >
        <div>
          <div>
            John Jones
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="new-cheque-button-wrapper"
  >
    <button
      onClick={[Function]}
    >
      Write another cheque
    </button>
  </div>
</div>
`;

exports[`Tag snapshot test 1`] = `
<div>
  <div
    className="cheque"
  >
    <div
      className="cheque-header"
    >
      <div
        className="cheque-logo-area"
      >
        <div
          className="logo-wrapper"
        >
          <div
            className="logo-icon"
          >
            <span
              aria-hidden={true}
              className="fa fa-hand-holding-usd"
            />
          </div>
          <div
            className="logo-bank-name"
          >
            First Bank of SmartCorp
          </div>
        </div>
        <div
          className="cheque-branch-name cheque-text"
        >
          Castle Hill Branch
        </div>
      </div>
      <div
        className="cheque-date-area"
      >
        <span
          className="cheque-text"
        >
          Date: 
        </span>
        <span
          className="cheque-date"
        >
          <div>
            <div>
              25/11/2019
            </div>
          </div>
        </span>
      </div>
    </div>
    <div
      className="cheque-body"
    >
      <div
        className="cheque-text-area"
      >
        <div
          className="cheque-text-payee-area"
        >
          <span
            className="cheque-text"
          >
            Pay:
          </span>
          <div
            className="cheque-payee-name"
          >
            <div>
              <div>
                Dan Kennedy
              </div>
            </div>
          </div>
        </div>
        <div
          className="cheque-text-amount-area"
        >
          <div
            className="cheque-text-amount"
          >
            <span>
              <div>
                <img
                  alt="loading"
                  className="loading-notice-spinner"
                  height="20px"
                  src="./images/loading.gif"
                  style={
                    Object {
                      "alignSelf": "center",
                      "display": "inline-block",
                    }
                  }
                  width="auto"
                />
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        className="cheque-numeric-amount-area"
      >
        <div
          className="cheque-numeric-amount"
        >
          <div>
            <span>
              $
            </span>
          </div>
          <div
            className="cheque-numeric-amount-text"
          >
            <div>
              <div>
                45
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cheque-footer"
    >
      <div
        className="cheque-signature-area"
      >
        <div>
          <div>
            John Jones
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="new-cheque-button-wrapper"
  >
    <button
      onClick={[Function]}
    >
      Write another cheque
    </button>
  </div>
</div>
`;
